### DISCUSSION TITLE
How do I destroy a module that has errors?

### DISCUSSION THREAD

**Discussion by yorinasub17** [gruntwork-io/knowledge-base#144] [Help] [answered]
I am trying to destroy a module, and am running into an error that is preventing me from destroy that appears to be a code issue. For example:

```
Error: Invalid index

  on .terraform/modules/server/modules/single-server/outputs.tf line 10, in output "public_ip":
  10:   value = var.attach_eip ? aws_eip.instance.*.public_ip[0] : aws_instance.instance.public_ip
    |----------------
    | aws_eip.instance is empty tuple

The given key does not identify an element in this collection value.Error: Invalid index

  on .terraform/modules/server/modules/single-server/outputs.tf line 10, in output "public_ip":
  10:   value = var.attach_eip ? aws_eip.instance.*.public_ip[0] : aws_instance.instance.public_ip
    |----------------
    | aws_eip.instance is empty tuple

The given key does not identify an element in this collection value.
```

Changing the `attach_eip` input has no effect. Is there anything I can do to destroy the resources in the state?

**Comment by gruntwork-discussions** [comment]
You must label this issue with at least one of the following labels before it can be answered:

* [r:terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)
* [r:terraform-aws-vpc](https://github.com/gruntwork-io/terraform-aws-vpc)
* [r:terraform-aws-architecture-catalog](https://github.com/gruntwork-io/terraform-aws-architecture-catalog)
* [r:terraform-aws-asg](https://github.com/gruntwork-io/terraform-aws-asg)
* [r:terraform-aws-beanstalk](https://github.com/gruntwork-io/terraform-aws-beanstalk)
* [r:terraform-aws-cache](https://github.com/gruntwork-io/terraform-aws-cache)
* [r:terraform-aws-ci](https://github.com/gruntwork-io/terraform-aws-ci)
* [r:terraform-aws-ci-pipeline-example](https://github.com/gruntwork-io/terraform-aws-ci-pipeline-example)
* [r:terraform-aws-cis-service-catalog](https://github.com/gruntwork-io/terraform-aws-cis-service-catalog)
* [r:terraform-aws-data-storage](https://github.com/gruntwork-io/terraform-aws-data-storage)
* [r:terraform-aws-ecs](https://github.com/gruntwork-io/terraform-aws-ecs)
* [r:terraform-aws-eks](https://github.com/gruntwork-io/terraform-aws-eks)
* [r:terraform-aws-elk](https://github.com/gruntwork-io/terraform-aws-elk)
* [r:terraform-aws-kafka](https://github.com/gruntwork-io/terraform-aws-kafka)
* [r:terraform-aws-lambda](https://github.com/gruntwork-io/terraform-aws-lambda)
* [r:terraform-aws-load-balancer](https://github.com/gruntwork-io/terraform-aws-load-balancer)
* [r:terraform-aws-messaging](https://github.com/gruntwork-io/terraform-aws-messaging)
* [r:terraform-aws-mongodb](https://github.com/gruntwork-io/terraform-aws-mongodb)
* [r:terraform-aws-monitoring](https://github.com/gruntwork-io/terraform-aws-monitoring)
* [r:terraform-aws-openvpn](https://github.com/gruntwork-io/terraform-aws-openvpn)
* [r:terraform-aws-sam](https://github.com/gruntwork-io/terraform-aws-sam)
* [r:terraform-aws-security](https://github.com/gruntwork-io/terraform-aws-security)
* [r:terraform-aws-server](https://github.com/gruntwork-io/terraform-aws-server)
* [r:terraform-aws-static-assets](https://github.com/gruntwork-io/terraform-aws-static-assets)
* [r:terraform-aws-zookeeper](https://github.com/gruntwork-io/terraform-aws-zookeeper)
* [r:terragrunt](https://github.com/gruntwork-io/terragrunt)
* [r:terratest](https://github.com/gruntwork-io/terratest)
* [r:repo-copier](https://github.com/gruntwork-io/repo-copier)


[Community Guidelines](https://github.com/gruntwork-io/knowledge-base/discussions/1)
[Getting help](https://gruntwork.io/support/)
[Gruntwork documentation](https://www.gruntwork.io)

**Comment by yorinasub17** [comment, marked as answer]
In some cases, you can avoid all code errors in `terraform` and trigger a `destroy` of all resources by creating a new `empty` module:

- Create a new module called `empty` with a single `main.tf` file that is empty.
- Ensure the `empty` module points to the state file of the module you are trying to destroy
    - If you are using `terragrunt`, update the `terraform.source` attribute in `terragrunt.hcl` to point to the empty module.
    - If you are using `terraform`, update the `empty` module to have a `terraform.backend` config that points to the state file of your original module.
- Now run `terragrunt init && terragrunt plan` (or `terraform init && terraform plan`). If you don't have nested providers (`provider` blocks nested in a terraform `module` call), this should create a valid plan to destroy all the resources. You can now run `terragrunt apply` or `terragrunt destroy` to destroy all the associated resources.
